Canary arm64 daily CI test is failing to create cluster-test.yaml

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The daily canary arm64 test is failing since December 2nd, it appears since #13246 was merged. I don't see how that PR is related to this error, but somehow triggered it. 

The deploy_cluster step is failing [here](https://github.com/rook/rook/blame/master/.github/workflows/daily-nightly-jobs.yml#L76-L79) when the cluster-test.yaml is created:
```
+ kubectl create -f cluster-test.yaml
cephcluster.ceph.rook.io/my-cluster created
Error from server (BadRequest): error when creating "cluster-test.yaml": CephBlockPool in version "v1" cannot be handled as a CephBlockPool: strict decoding error: unknown field "spec.healthCheck"
Error: Process completed with exit code 1.
```

This error seems to indicate that the yaml is malformed because it's complaining about the spec.healthCheck when creating the CephBlockPool, even though spec.healthCheck is actually part of the CephCluster. 

I wonder if the old version of [yq 3.4.1](https://github.com/rook/rook/blame/master/.github/workflows/daily-nightly-jobs.yml#L55-L58) is not sufficient anymore, or if updating it will help fix this issue. There are no errors in the CI output that indicates yq failed, it's only a suspicion. 

This canary arm64 is the only test with this issue, thus it seems related to the yq that this specific test uses.

**Expected behavior:**
The daily canary arm64 test is expected to pass

--- a/.github/workflows/daily-nightly-jobs.yml
+++ b/.github/workflows/daily-nightly-jobs.yml
@@ -73,9 +73,9 @@ jobs:
           # Use the official build images for the nightly arm tests instead of rebuilding
           export USE_LOCAL_BUILD=false
           # removing liveness probes since the env is slow and the probe is killing the daemons
-          yq write -d1 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.mon.disabled" true
-          yq write -d1 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.mgr.disabled" true
-          yq write -d1 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.osd.disabled" true
+          yq write -d0 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.mon.disabled" true
+          yq write -d0 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.mgr.disabled" true
+          yq write -d0 -i deploy/examples/cluster-test.yaml "spec.healthCheck.livenessProbe.osd.disabled" true
           tests/scripts/github-action-helper.sh deploy_cluster
           # there are no package for arm64 nfs-ganesha
           kubectl delete -f deploy/examples/nfs-test.yaml

